scripts/gke_integration_test.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+# Copyright 2017 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Setup environment for CI with GKE.
+# This include:
+# - Building a test application into a docker image
+# - Upload the image to Google Container Registry
+# - Create a Kubernetes cluster on GKE
+# - Deploy the application
+# - Run the integration test
+set -e
+
+readonly dir=$(dirname $0)
+readonly projectRoot="$dir/.."
+readonly testAppDir="$projectRoot/test-application"
+readonly deployDir="$testAppDir/target/deploy"
+
+readonly projectName=$(gcloud info \
+                | awk '/^Project: / { print $2 }' \
+                | sed 's/\[//'  \
+                | sed 's/\]//')
+readonly imageName="openjdk-gke-integration"
+readonly imageUrl="gcr.io/$projectName/$imageName"
+readonly clusterName="openjdk-gke-integration"
+
+readonly imageUnderTest=$1
+if [[ -z "$imageUnderTest" ]]; then
+  echo "Usage: ${0} <image_under_test>"
+  exit 1
+fi
+
+# build the test app
+pushd ${testAppDir}
+mvn clean install
+popd
+
+# deploy to app engine
+pushd ${deployDir}
+export STAGING_IMAGE=${imageUnderTest}
+envsubst < "Dockerfile.in" > "Dockerfile"
+export TESTED_IMAGE=${imageUrl}
+envsubst < "openjdk-spring-boot.yaml.in" > "openjdk-spring-boot.yaml"
+
+echo "Deploying image to Google Container Engine..."
+docker build -t "$imageName" .
+docker tag "$imageName" "$imageUrl"
+gcloud docker -- push gcr.io/${projectName}/openjdk-gke-integration
```

Comment:
It's worth thinking about. One reason why it might not be a good idea - these tests require using a custom base image, which would require overriding the default builder to point to the base image under test. 

I think if we want to test openjdk runtime candidates with the builder pipeline, that would belong in the builder's integration tests.